src/client/vite.config.ts

```diff
@@ -270,22 +261,18 @@ const setConfig: (env: ConfigEnv) => UserConfigExport = ({ mode }) => {
   const isDev = mode === "development"
   const viteBaseUrl = isDev ? "/" : getBaseUrl(false)
 
-  const plugins: any[] = [react()]
+  const plugins: any[] = [react()] // stays as any[] as WB injectManifest does not return PluginOption
 
   if (!isDev) {
-    plugins.push(customPreloadPlugin())
+    plugins.push(splitVendorChunkPlugin(), customPreloadPlugin())
   }
 
-  const TARGET = process.env.TARGET || "web"
+  const buildTarget: BuildTarget = (process.env.TARGET as BuildTarget) || "web"
 
```

Comment:
very minor, orthogonal to the changes in this PR, and not a blocker at all, but FYI it seems the idiomatic way to apply certain plugins in build vs serve (e.g. development mode) is like this:
https://vitejs.dev/guide/api-plugin.html#conditional-application

Reply:
actually probably not relevant, I realized it only applies to line 267, the other if statements would still be necessary

Reply:
I see .. that is a bit buried in the plugin conf ... I prefer our existing approach, actually